examples/angular/src/pages/ui/components/authenticator/sign-up/sign-up.component.html

```diff
@@ -1 +1,10 @@
-<amplify-authenticator></amplify-authenticator>
+<amplify-authenticator>
+  <ng-template
+    amplifyOverride="authenticated"
+    let-user="user"
+    let-signOut="signOut"
```

Comment:
In the future, if we have a react hook like `useAuthenticator()` that returns a `{ user, signOut }`, could this template become:

```html
<amplify-authenticator>
  <ng-template amplifyOverride="authenticated" let-context="authenticator">
    <h2>Welcome, {{ authenticator.user.username }}!</h2>
    <button (click)="authenticator.signOut($event)">Sign Out</button>
  </ng-template>
</amplify-authenticator>
```

I'm looking to understand if having N `let-*` attributes is common or not in Angular :)

Reply:
This is fairly new, so this is not so common in the Angular wild. I could find `let-` usage in angular-bootstrap: https://ng-bootstrap.github.io/#/components/datepicker/overview#day-template.

Yup that's totally possible too! Or even: 

```html
<amplify-authenticator>
  <ng-template amplifyOverride="authenticated" let-authenticator>
    <h2>Welcome, {{ authenticator.user.username }}!</h2>
    <button (click)="authenticator.signOut($event)">Sign Out</button>
  </ng-template>
</amplify-authenticator>
```
This looks like a great idea actually! I'll consider this.

For more finer control, I still expect customers to use `AuthenticatorService` directly instead of `let-` so that they have programmatic control over these files. `let-` is more for convenience for fetching and using related objects directly. I can write a quick use case document after this pr.

Reply:
I'll put this into our facade task after qa. But that above code snippet looks promising for in-template use cases.